src/app/app.component.ts

```diff
@@ -114,46 +114,37 @@ export class AppComponent implements OnInit, OnDestroy {
 			)
 			.subscribe();
 
-		const suggestions$ = this.addPackage.valueChanges.pipe(
-			debounceTime(200),
-			switchMap((query) => {
-				if (!query) {
-					return of([]);
-				}
-
-				return this.apiService.getSuggestions(query);
-			}),
-			withLatestFrom(this.packageNames.observable$),
-			map(([suggestions, existingPackageNames]) =>
-				suggestions.filter((suggestion) => !existingPackageNames.includes(suggestion))
-			)
-		);
+		// npm library suggestions on user input
+		const suggestions$ = this.searchLibraryOnInputChange();
 
+		// autocomplete options to show package names for the user
 		this.autocompleteOptions$ = combineLatest([
-			// All possible npm package names (based on local storage)
-			this.autocompletePackageNames.observable$,
 			// All suggestions
-			suggestions$,
-			// Already selected package names
-			this.packageNames.observable$,
+			suggestions$.pipe(startWith([])),
+			// packages that already have been loaded
+			this.packageNames.observable$.pipe(startWith(this.packageNames.getValue())),
 			// Partial npm package name to filter options
 			this.addPackage.valueChanges.pipe(startWith('')),
 		]).pipe(
-			map(([autocompletePackageNames, suggestions, packageNames, query]) =>
-				this.getAutocompleteOptions([...new Set([...autocompletePackageNames, ...suggestions])], packageNames, query)
+			map(([suggestions, alreadyLoadedPackageName, query]) =>
+				this.getAutocompleteOptions(
+					[...new Set([...(this.autocompletePackageNames.getValue() ?? []), ...suggestions])],
+					alreadyLoadedPackageName,
```

Comment:
I think `alreadyLoadedPackageName` should be plural. Isn't it an array from `this.packageNames.observable$`?

`alreadyLoadedPackageName` -> `alreadyLoadedPackageNames`

Reply:
changed in last commit